src/components/images/draw.js

```diff
@@ -0,0 +1,125 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+var Drawing = require('../drawing');
+var Axes = require('../../plots/cartesian/axes');
+
+module.exports = function draw(gd) {
+
+    var fullLayout = gd._fullLayout,
+        imageDataAbove = [],
+        imageDataSubplot = [],
+        imageDataBelow = [];
+
+    if(!fullLayout.images) return;
+
+
+    // Sort into top, subplot, and bottom layers
+    for(var i = 0; i < fullLayout.images.length; i++) {
+        var img = fullLayout.images[i];
```

Comment:
But, that means that a `layout` with:

```js
images: [{}]
```

results in no images shown on the graph - which is not consistent with `annotations` and `shapes`, but perhaps the better default for `images`.

Unlike corresponding `annotations` and `shapes` attributes, there's no suitable `source` defaults, except maybe the plotly logo?